tracing/trace_state.hh

```diff
@@ -239,7 +247,11 @@ private:
      */
     void begin() {
         std::atomic_signal_fence(std::memory_order_seq_cst);
-        _start = elapsed_clock::now();
+
+        // Set the start point unless already inherited from the primary session
+        if (_start == elapsed_clock::time_point::min()) {
+            _start = elapsed_clock::now();
+        }
```

Comment:
corner case two: if clocks on the nodes are badly unsynchronized and `_start` appears to be in the future - also prefer to use `elapsed_clock::now()`

Reply:
I have rounded your corner cases with my std::min(start, local_clock::now())